### ISSUE TITLE
NPE in GzipRequestInterceptor

### ISSUE THREAD

**Issue by Mordag** [mapbox/mapbox-events-android#416]
**Configuration**
 - SDK Version mapbox-android-sdk using version 8.0.0
 - Android OS Version 8.0.0
 - Device model(s): Google Pixel XL

**Logging**
```
2019-06-07 09:31:39.205 19772-19835/com.my.app.qa E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
   Process: com.my.app.qa, PID: 19772
   java.lang.NullPointerException: Attempt to invoke virtual method 'okhttp3.Route okhttp3.internal.connection.RealConnection.route()' on a null object reference
       at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.java:233)
       at okhttp3.internal.connection.RealConnection.createTunnel(RealConnection.java:400)
       at okhttp3.internal.connection.RealConnection.connectTunnel(RealConnection.java:236)
       at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:177)
       at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.java:224)
       at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.java:107)
       at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.java:87)
       at okhttp3.internal.connection.Transmitter.newExchange(Transmitter.java:169)
       at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:41)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
       at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
       at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
       at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
       at com.mapbox.android.telemetry.GzipRequestInterceptor.intercept(GzipRequestInterceptor.java:36)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
       at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:221)
       at okhttp3.RealCall$AsyncCall.execute(RealCall.java:172)
       at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)
```

**Comment by andrlee** [comment]
@Mordag thanks for reporting this issue. can you share with us steps how we can reproduce it? is this only reproducible with `android 8.0`?

**Comment by andrlee** [comment]
@Mordag based on my brief research this is a known issue with `okhttp`: https://github.com/square/okhttp/issues/4915. We unfortunately can't upgrade to `3.14.2` since it requires Android 5.0 and Java 8. I will request Jesse to cherry pick the fix to `3.12.x` that's the best i can do atm

**Comment by Mordag** [comment]
@andrlee Thank you for your quick response. I hope okhttp will release a new patch soon.
